result

ADIGrimm opened the pull request.

ADIGrimm committed:
result

Elena-Bruyako commented on pom.xml:
```diff
@@ -95,6 +95,12 @@
 			<artifactId>liquibase-maven-plugin</artifactId>
 			<version>${liquibase.version}</version>
 		</dependency>
+		<dependency>
+			<groupId>org.hibernate.validator</groupId>
+			<artifactId>hibernate-validator</artifactId>
+			<version>8.0.1.Final</version>
+		</dependency>
```
```suggestion
		<dependency>
                       <groupId>org.springframework.boot</groupId>
                       <artifactId>spring-boot-starter-validation</artifactId>
                </dependency>
```

Elena-Bruyako commented on src/main/java/online/bookstore/dto/CreateBookRequestDto.java:
```diff
@@ -1,13 +1,20 @@
     private String title;
+    @NotBlank
     private String author;
+    @ISBN
```
```suggestion
    @ISBN
    @NotBlank
```

Elena-Bruyako replied:
You have your own validation, let's use it or remove

Elena-Bruyako commented on src/main/java/online/bookstore/dto/CreateBookRequestDto.java:
```diff
@@ -1,13 +1,20 @@
     private String author;
+    @ISBN
     private String isbn;
+    @Min(0)
```
```suggestion
    @Min(0)
    @NotNull
```

Elena-Bruyako commented on src/main/java/online/bookstore/exception/CustomGlobalExceptionHandler.java:
```diff
@@ -0,0 +1,64 @@
+                .map(this::getErrorMessage)
+                .toList();
+        body.put("errors", errors);
+        return new ResponseEntity<>(body, headers, status);
```
```suggestion
        return new ResponseEntity<>(body, headers, HttpStatus.BAD_REQUEST);
```

Elena-Bruyako commented on src/main/java/online/bookstore/exception/CustomGlobalExceptionHandler.java:
```diff
@@ -0,0 +1,64 @@
+    public ResponseEntity<Object> handleDataProcessingException(DataProcessingException ex) {
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put("timestamp", LocalDateTime.now());
+        body.put("status", HttpStatus.INTERNAL_SERVER_ERROR);
```
here you use INTERNAL_SERVER_ERROR and on 53 line NOT_FOUND
let's use correct status

Elena-Bruyako reviewed (changes requested):
see comments

ADIGrimm committed:
dependency hibernate validator now spring boot starter validation
fixed annotations in CreateBookRequestDto
fixed return in handleMethodArgumentNotValid()
fixed status in handleDataProcessingException()

Elena-Bruyako commented on src/main/java/online/bookstore/exception/CustomGlobalExceptionHandler.java:
```diff
@@ -0,0 +1,64 @@
+        return new ResponseEntity<>(body, HttpStatus.NOT_FOUND);
+    }
+
+    @ExceptionHandler(DataProcessingException.class)
```
You don't throw this exception in code, so you can remove handling

Elena-Bruyako reviewed (changes requested):
one comment

ADIGrimm committed:
removed unnecessary handler
